src/main/java/Duke.java

```diff
@@ -1,10 +1,215 @@
+                list.add(task);
+                System.out.println(text);
+            }
+            sn.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("There are no tasks as of now!");
+        } catch (DukeException d) {
+            System.out.println(d.toString());
+        }
+
+        Scanner scn = new Scanner(System.in);
+        while (true) {
+
+            String input = scn.nextLine();
+
+            if (input.startsWith("mark")) {
+
+                String index = input.split(" ", 2)[1];
+                Task task = list.get(Integer.parseInt(index) - 1);
+                task.markAsDone();
+                System.out.println("Nice! I've marked this task as done:\n" + task.toString());
+
+            } else if (input.startsWith("unmark")) {
+
+                String index = input.split(" ", 2)[1];
+                Task task = list.get(Integer.parseInt(index) - 1);
+                task.unmark();
+                System.out.println("OK, I've marked this task as not done yet:\n" + task.toString());
+
+            } else if (Objects.equals(input.toLowerCase(), "bye")) {
+
+                int len = list.size();
+                String text = "";
+                for (int i = 0; i < len; i++) {
+                    text += list.get(i).toString();
+                    text += "\n";
+                }
+
+                try {
+                    writeToFile("data/duke.txt", text);
+                } catch (IOException e) {
+                    System.out.println("Error saving file: " + e.getMessage());
+                }
+
+                System.out.println("Bye. Hope to see you again soon!");
+                scn.close();
+                return;
+
+            } else if (Objects.equals(input.toLowerCase(), "list")) {
+
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 1; i <= list.size(); i++) {
+                    Task task = list.get(i - 1);
+                    System.out.println(i + "." + task.toString());
+                }
+
+            } else if (input.startsWith("delete")) {
+                String index = input.split(" ", 2)[1];
+                Task task = list.get(Integer.parseInt(index) - 1);
+                list.remove(Integer.parseInt(index) - 1);
+                System.out.println("Noted. I've removed this task:\n" + task.toString() + "\nNow you have "
+                        + list.size() + " tasks in the list.");
+            } else {
+                try {
+                    if (input.contains("todo")) {
+                        String description = input.replace("todo", "");
+                        Todo todo = new Todo(description);
+                        list.add(todo);
+                        System.out.println("Got it. I've added this task:\n" + todo.toString() + "\nNow you have "
+                                + list.size() + " tasks in the list.");
+                    } else if (input.contains("deadline")) {
```

Comment:
Not sure if my logic is wrong but would it be better to check if input starts with deadline instead of contain? If somehow the input is "deadline make a todo app /by 2023/02/01" does it cause the code to excercute the previous if statement block of todo?

```suggestion
                    } else if (input.startsWith("deadline")) {